generator/src/main/java/com/scottlogic/deg/generator/generation/string/RegexStringGenerator.java

```diff
@@ -302,7 +302,8 @@ void updateMatchedStringIdx() {
                 for (Node childNode : nextNodes) {
                     childNode.updateMatchedStringIdx();
                     long childNbrChar = childNode.getMatchedStringIdx();
-                    matchedStringIdx += nbrChar * childNbrChar;
+                    long newNbrChar = (childNbrChar == 0 || Long.MAX_VALUE / childNbrChar > nbrChar) ? nbrChar * childNbrChar : nbrChar;
```

Comment:
@afroggattsl do you know what Nbr stands for?

Reply:
I assumed Number but interpreted as child character count, new character count etc.  And Assuming its to do with working out how many different combincation of chars are allowed as you increment size of string.

Reply:
Can you move some of the new calculations to a well named method.

e.g. so it reads something like

if (wouldOverflow(x, y) {
  matched = 
else
  matched =